pkg/iss/leaderselectionpolicy/leaderselectionpolicy_test.go

```diff
@@ -143,9 +129,21 @@ func TestBlacklistLeaderPolicy_EmptySuspected(t *testing.T) {
 	assert.IsType(t, &BlacklistLeaderPolicy{}, reconstitutedPolicy)
 	reconstitutedBlacklistPolicy := reconstitutedPolicy.(*BlacklistLeaderPolicy)
 	assert.Equal(t, policy.Membership, reconstitutedBlacklistPolicy.Membership)
-	assert.Equal(t, policy.MinLeaders, reconstitutedBlacklistPolicy.MinLeaders)
 	assert.Equal(t, policy.Suspected, reconstitutedBlacklistPolicy.Suspected)
 	policyLeaders := policy.Leaders()
 	reconstitutedPolicyLeaders := policy.Leaders()
 	assert.Equal(t, policyLeaders, reconstitutedPolicyLeaders)
 }
+
+func testMembership(nodeIDs []types.NodeID) *trantorpbtypes.Membership {
+	m := trantorpbtypes.Membership{Nodes: map[types.NodeID]*trantorpbtypes.NodeIdentity{}}
+	for _, nodeID := range nodeIDs {
+		m.Nodes[nodeID] = &trantorpbtypes.NodeIdentity{
+			Id:     nodeID,
+			Addr:   "",
+			Key:    nil,
+			Weight: 1,
```

Comment:
As a note: having (not necessarily in the lsp package) tests for variable weight sizes would not be bad.

Reply:
Yes, once we define what weights are considered invalid, if any. I thought about that a bit, and realized that I don't see any fundamental reason to disallow any set of weights. The only exception is all weights in a membership being zero, which we probably do need to treat as a special case. I see 2 options here:
1. Declare such a membership invalid and complain / ignore it / crash / ...
2. Treat it as a special case and pretend that all weights are 1. I can imagine this being convenient for implementations that simply don't want to deal with any vote weights - it automatically falls back into counting nodes. However, it introduces additional complexity not just in code, but especially in the semantics.